src/MOL_discretization.jl

```diff
@@ -1,22 +1,224 @@
+
+    no_iv = size(pdesys.domain,1)
+    domain = []
+    dx = []
+    X = []
+    xx = []
+    for i = 1:no_iv-1
+        domain = vcat(domain,pdesys.domain[i+1].domain)
+        dx = vcat(dx,discretization.dxs)
+        X = vcat(X,domain[i].lower:dx[i]:domain[i].upper)
+        xx = vcat(xx,size(X,1)-2)
+    end
+
+    # TODO: specify order for each derivative
+    order = discretization.order
+
+    ### Calculate discretization expression ####################################
+    # The discretization is an expression which is then evaluated
+    # in the ODE function (f)
+
+    # TODO: improve the code below using index arrays instead of Dicts?
+    lhs_nonderiv_depvars = Dict()
+    lhs_deriv_depvars = Dict()
+    discretization = Dict()
+    # if there is only one equation
+    if pdesys.eq isa Equation
+        var = pdesys.eq.lhs.args[1].op
+        discretization[var] = discretize_2( pdesys.eq.rhs,0,order,dx[1],
+                                            [var],Dict())
+
+    # if there are many equations (pdesys.eq isa Array)
+    else
+        # Store 'non-derived' dependent variables (e.g. v(t,x)=t*x)
+        # and 'derived' dependent variables (e.g. Dt(u(t,x)))
+        n_eqs = size(pdesys.eq,1)
+        for i = 1:n_eqs
+            input = pdesys.eq[i].lhs
+            if input.op isa Variable
+                var = input.op
+                lhs_nonderiv_depvars[var] = pdesys.eq[i].rhs
+            else #var isa Differential
+                var = input.args[1].op
+                lhs_deriv_depvars[var] = pdesys.eq[i].rhs
+            end
+        end
+
+        # Calc. coeff. matrix for each differential equation
+        lhs_deriv_depvars_arr = collect(keys(lhs_deriv_depvars))
+        for (var,rhs) in lhs_deriv_depvars
+            discretization[var] = discretize_2( rhs,0,order,dx[1],
+                                                lhs_deriv_depvars_arr,
+                                                lhs_nonderiv_depvars)
+        end
+    end
+
+    ### Get boundary conditions ################################################
+    # TODO: generalize to N equations
+    lhs_deriv_depvars_bcs = get_bcs(pdesys.bcs,tdomain,domain[1])
+    t = 0.0
+    interior = domain[1].lower+dx[1]:dx[1]:domain[1].upper-dx[1]
+    u0 = Array{Float64}(undef,length(interior),length(discretization))
+    Q = Array{RobinBC}(undef,length(discretization))
+    
+    i = 1
+    for var in keys(discretization)
+        bcs = lhs_deriv_depvars_bcs[var]
+
+        g = eval(:((x,t) -> $(bcs[1])))
+        u0[:,i] = @eval $g.($interior,$t)
+        
+        u_x0 = eval(bcs[2])
+        u_x1 = eval(bcs[3])
+        Q[i] = DirichletBC(u_x0,u_x1)
+
+        i = i+1
+
+    end
+
+
+    Qu = Array{Float64}(undef,xx[1]+2,length(discretization))
+
+    ### Define the discretized PDE as an ODE function ##########################
+    function f(du,u,p,t)
+        for j in 1:length(discretization)
+            Qu[:,j] = Q[j]*u[:,j]            

```

Comment:
views

Reply:
I replaced it with the code below, but I am not sure if this is how it should be done.
```Qu = [view(Q[j]*u[:,j],:) for j=1:length(discretization)]```